components/app.js

```diff
@@ -0,0 +1,348 @@
+}
+
+/**
+ * Identify how many users ordered both food item & beverage
+ */
+function getUsersWithFoodAndBeverage() {
+    const orders = ordersData;
+
+    const usersWithFoodAndBeverage = new Set(
+        orders.filter(order => (
+            order.item && order.item.trim() !== '' && 
+            order.drink && order.drink.trim() !== '' 
+        )).map(order => order.userId)
+    );
+    
+    const numberOfUsers = usersWithFoodAndBeverage.size;
+    console.log(`Number of users who ordered both food and beverage: ${numberOfUsers}`);
+}
+
+/**
+ * Given any restaurant name & item name identify how many sold
+ */
+function getSoldQuantityInfo() {
+    const orders = ordersData;
+    const restaurantName = prompt('Please enter the restaurant name'); //Honest John Pizza
+    const itemName = prompt('Please enter the food item name'); //cheese pizza
+
+    if(restaurantName.length !== 0 || itemName.length !== 0){
+        const totalSold = orders.reduce((total, order) => {
+            if (order.restaurant_name === restaurantName && order.item === itemName) {
+                total += order.quantity;
+            }
+            return total;
+        }, 0);
+    
+        console.log(`Total ${itemName} sold at ${restaurantName}: ${totalSold}`);
+    } else {
+        console.log('Invalid input. Please enter valid restaurant name or item name');
+    }
+    
+}
+
+/**
+ * Which all restaurants serve given beverage or food
+ */
+function getRestaurantInfo() {
+    const restaurantData = restaurantsData;
+    const itemToFind = prompt('Please enter the beverage/food item name'); 
+
+    console.log(itemToFind);
+    if (!itemToFind.length !== 0) {
+        const restaurantsServingItem = restaurantData
+        .filter(restaurant => restaurant.food.includes(itemToFind) || restaurant.beverages.includes(itemToFind))
+        .map(restaurant => restaurant.name);
+
+        console.log(`Restaurants serving ${itemToFind}:`);
+        console.log(restaurantsServingItem);
+    } else {
+        console.log('Invalid input. Please enter valid beverage/food item name');
+    }
+}
+
+/**
+ * Sort orders by date
+ */
+function sortOrdersByDate() {
+    const userInput = prompt('Do you want to sort in ASC/DESC?');
+
+    const orders = ordersData;
+    const ordersWithUpdatedDate = orders.map(obj => ({ ...obj, date: new Date(obj.date) }));
+
+    let sortedOrders;
+    if (userInput === 'ASC' || userInput === 'DESC') {
+        const sortOrder = userInput === 'ASC' ? 1 : -1;
+        sortedOrders = ordersWithUpdatedDate.sort((prevOrder, nextOrder) => {
+            return sortOrder * (prevOrder.date - nextOrder.date);
+        });
+        const orderMap = new Map(sortedOrders.map((order, index) => [order.id, index]));
+        console.log(`Sorted Orders in ${userInput === 'ASC' ? 'Ascending' : 'Descending'}`, 
+                    orders.sort((prevOrder, nextOrder) => orderMap.get(prevOrder.id) - orderMap.get(nextOrder.id)));
```

Comment:
could you let me know why sort is used here

Reply:
This was done to sort the updatedOrders where the date was modified. 

I have optimized the code without updating the list and modifying the date within the sort function